### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by JackTemaki**
This mainly adds empty docstrings and an empty sphinx documentation.

There is a small add-on for writing the state-tying in there, I apologize for this being here in the wrong PR.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# -- Path setup --------------------------------------------------------------
+
+# If extensions (or modules to document with autodoc) are in another directory,
+# add these directories to sys.path here. If the directory is relative to the
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+#
+import os
+import sys
+
+# code is used for automatic api generati code is used for automatic api generationon
```
Comment (michelwi) [inline docs/conf.py:16]:
```suggestion
# code is used for automatic api generation
```

**Review by michelwi — COMMENTED** [review]
I like your endeavor to create a documentation for the pipeline.

Should this PR inform us, that a documentation is under construction, or do you plan to complete all the TODOs of this PR before it lands?

**Comment by JackTemaki** [comment]
> I like your endeavor to create a documentation for the pipeline.
> 
> Should this PR inform us, that a documentation is under construction, or do you plan to complete all the TODOs of this PR before it lands?

The first one, @christophmluscher and I discussed that we want to merge even the under-construction code into the master, so that it is more easy to work and test with it. And also I have no capabilities/knowledge to actually fill all of those TODOs....

So I hope this stuff gets filled during the process of finishing the code and testing.

**Comment by christophmluscher** [comment]
> So I hope this stuff gets filled during the process of finishing the code and testing.

I already planned to help with Doc :)

**Comment by JackTemaki** [comment]
@michelwi @christophmluscher can I merge this or not?

**Review by michelwi — APPROVED** [review]

**Review by christophmluscher — APPROVED** [review]

**Pull request merged**